src/main/java/com/vinylteam/vinyl/dao/jdbc/JdbcUserDao.java

```diff
@@ -150,4 +153,24 @@ public boolean update(String email, User user) {
         return Optional.ofNullable(user);
     }
 
+    @Override
+    public Optional<User> findById(long id) {
+        User user = null;
+        try (Connection connection = dataSource.getConnection();
+             PreparedStatement findByEmailStatement = connection.prepareStatement(FIND_BY_ID)) {
+            findByEmailStatement.setLong(1, id);
+            try (ResultSet resultSet = findByEmailStatement.executeQuery()) {
+                if (resultSet.next()) {
+                    user = userRowMapper.mapRow(resultSet);
+                    if (resultSet.next()) {
+                        throw new RuntimeException("More than one user was found for id");
+                    }
+                }
+            }
+        } catch (SQLException e) {
+            throw new RuntimeException(e);
```

Comment:
@SneakThrows

Reply:
I will add logs and therefore refuse to use this annotation. Thanks for the comment

Reply:
I think logs is unnecessary here, and @SneakyThrows is much better solution